chore: remove webpack from unit test process

itsjgf opened the pull request:
<!-- Reviewable:start -->

Closes #110

<!-- Reviewable:end -->

joshdmiller commented on .babelrc:
```diff
@@ -0,0 +1,9 @@
+{
```
Why was this moved to a `.babelrc` from the `package.json` entry?

joshdmiller commented on webpack.config.js:
```diff
@@ -16,6 +17,9 @@ var shared = {
       STORYSHOP_API_URI: dev ? "'http://localhost:9999'" : "'http://api.storyshopapp.com'",
     }),
+    new CleanWebpackPlugin(['build'], {
+      exclude: ['index.html']
```
In order to avoid a `cp` command, we removed a valid `.gitignore`, altered all the build scripts, added a cleaning script, and excluded the html from the clean. And this will require developers to try and remember not to commit the rest of the files in build. I feel things may be getting a little hacky at this point.

If the only reason for all these changes is to support Windows, I'd just favour the Windows users installing a virtual machine because it feels like we're making a lot of compromises. What do you think?

joshdmiller commented on package.json:
```diff
@@ -89,7 +86,9 @@
-    "whatwg-fetch": "^0.11.0"
+    "whatwg-fetch": "^0.11.0",
+    "better-npm-run": "0.0.11",
+    "clean-webpack-plugin": "^0.1.10"
```
Looks like you forgot to save the `babel-module-resolver` dependency. :smile:

joshdmiller reviewed (changes requested):
I'm not sold on the "cross-platform" (read: for Windows) changes, but look forward to hearing more from you on that subject. The use of `babel-module-resolver` is good, but I have a couple of questions on that too.

:boom:

itsjgf committed:
chore: remove webpack from unit test process

chore: merge webpack configuration

chore: improve test:unit script

chore: move specs.js from src to conf

Closes #110

joshdmiller commented on webpack.config.js:
```diff
@@ -62,22 +62,6 @@ module.exports = [
-      filename: 'specs.js',
-    },
-  }),
+  })
```
I think we needn't export the array anymore. We should also be able to merge the `shared` config directly into this one; it makes no sense factored out anymore.

joshdmiller commented on package.json:
```diff
@@ -7,9 +7,8 @@
     "build:html": "mkdir -p build && cp index.html build/",
     "build": "npm run build:html && npm run build:webpack",
-    "test:unit": "tape build/specs.js | faucet",
+    "test:unit": "tape -r babel-register ./src/specs.js ./src/**/spec.js | faucet",
```
I think we should account for allowing multiple spec files in a directory. For example, if we were to add tests for our reducers, they'd likely be at `src/store/reducers/auth.spec.js` and  `src/store/reducers/meta.spec.js`, which would not be picked up by the pattern. Perhaps `src/**/*spec.js` is a better pattern.

Also, we may want to wrap the glob in quotes. Not every environment will accept the double asterisk globbing pattern, but if we wrap it in quotes, it will be resolved on the node side.

joshdmiller reviewed (changes requested):
Looking pretty good! Two minor changes.

joshdmiller reviewed (changes requested):
Looking pretty good. Two last things. Let's squash all those commits into one with a `rebase -i`. And if we're only using `src/specs.js` now to set up a variable for the tests, it is no longer a source file. Let's move it to `conf` and require it with `tape` using a second `-r`.

joshdmiller commented:
lgtm. Let's add the `Closes #110` to the commit and call it done!
